packages/realm/src/app-services/SyncSession.ts

```diff
@@ -100,6 +106,83 @@ export function toBindingErrorHandler(onError: ErrorCallback) {
   };
 }
 
+/** @internal */
+export function toBindingErrorHandlerWithOnManual(
+  onError: ErrorCallback | undefined,
+  onManual: ClientResetFallbackCallback | undefined,
+) {
+  if (!onError && !onManual) {
+    throw new Error("need either onError or onManual");
+  }
+  if (onError && onManual) {
+    return (sessionInternal: binding.SyncSession, bindingError: binding.SyncError) => {
+      const session = new SyncSession(sessionInternal);
+      const error = fromBindingSyncError(bindingError);
+      if (error instanceof ClientResetError) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        onManual(session, error.config.path!);
+      } else {
+        onError(session, error);
+      }
+      session.resetInternal();
+    };
+  }
+  if (onError) {
+    return (sessionInternal: binding.SyncSession, bindingError: binding.SyncError) => {
+      const session = new SyncSession(sessionInternal);
+      const error = fromBindingSyncError(bindingError);
+      if (error instanceof ClientResetError) {
+        onError(session, error);
+      }
+      session.resetInternal();
+    };
+  }
+  // onManual must be true here
```

Comment:
You can see this but I doubt the TypeScript compiler can. And I think it answers your question why `onManual` isn't pruned.